routes/conversation.js

```diff
@@ -255,6 +257,74 @@ exports.leave = function (req, res){
 	});
 };
 
+/**
+ * Action when a user archives a conversation.
+ */
+exports.archive = function (req, res){
+	if (!req.user || !req.user._id){
+		resError(res, "Access denied.", "/error");
+		return;
+	}
+
+	// Search through the user's conversation list until the conversation with the correct
+	// id is found. 
+	User.findById( req.user._id, function (err, user){
+		if (err){
+			resError(res, "Could not find the user.");
+			return;
+		}
+
+		// Save the id of the conversation that we want to archive.
+		var targetConversationId = JSON.stringify(req.body.conversationId);
+
+		// Find the index at which the target conversation exists amongst all of the user's conversations.
+		var conversationIds = user.userConversations.map(function(p) {return p.conversation});
+		var conversationIdStrings = conversationIds.map(function(id) {return JSON.stringify(id)});
+		var foundIndex = conversationIdStrings.indexOf(targetConversationId);
+
+		// Modify the hallOfFame boolean associated with the conversation to true if it exists.
+		if (foundIndex !== -1) {
+			user.userConversations[foundIndex].hallOfFame = true;
+
+			// Save the modified user object.
+			user.save(function(err){
+				if (err) return resError(res, "Could not save user to database.");
+
+				// If there are no errors, proceed to update the isThrilled boolean of the user
+				// in the conversation object to true and save this object.
+				Conversation.findById(targetConversationId, function(err, convo){
```

Comment:
When doing a database query, it seems like it wants to use the original ID object as part of the query.  `targetConversationId` is no longer the object that Mongo expects because we used `JSON.stringify` to turn it into a string on line 278.  Thus, `targetConversationId` should probably be changed to the original id, `req.body.conversationId`.

I only know about this because I ran into an error during the summer that depended on it.  Took me forever to figure it out!